client/js/components/create_form.jsx

```diff
@@ -0,0 +1,250 @@
+var React = require("react");
+var _ = require("lodash");
+var $ = require("jquery");
+var router;
+var AppStore = require("../store/app_store");
+
+var CreateForm = React.createClass ({
+  componentWillMount: function() {
+    router = this.props.router;
+  },
+
+  componentDidMount: function() {
+    AppStore.addChangeListener(this.forceUpdate);
+  },
+
+  componentWillUnmount: function() {
+    AppStore.removeChangeListener(this.forceUpdate);
+  },
+
+  trigger: function(){
+    this.forceUpdate();
+  },
+
+  createFormSchema: function(event){
```

Comment:
This function should be in the app store. And then we could call the router in a callback.